packages/connect/src/ui.ts

```diff
@@ -0,0 +1,41 @@
+import { authenticate, AuthOptions, FinishedData } from './auth';
+import { defineCustomElements } from '@blockstack/connect-ui';
+
+export const showBlockstackConnect = (authOptions: AuthOptions) => {
+  defineCustomElements();
+  const element = document.createElement('connect-modal');
+  element.authOptions = authOptions;
+  document.body.appendChild(element);
+  const finishedWrapper = (finishedData: FinishedData) => {
+    element.remove();
+    const callback = authOptions.onFinish || authOptions.finished;
+    if (callback) {
+      callback(finishedData);
+    }
+  };
+  element.addEventListener('signUp', () => {
+    authenticate({
+      ...authOptions,
+      sendToSignIn: false,
+      onFinish: finishedWrapper,
+    });
+  });
+  element.addEventListener('signIn', () => {
+    authenticate({
+      ...authOptions,
+      sendToSignIn: true,
+      onFinish: finishedWrapper,
+    });
+  });
+  const handleEsc = function (ev: KeyboardEvent) {
+    if (ev.key === 'Escape') {
+      document.removeEventListener('keydown', handleEsc);
+      element.remove();
+    }
+  };
+  element.addEventListener('closeModal', () => {
```

Comment:
Event might be better named `onModalClose` / `modalclose`

Reply:
Can pass an argument `@Event({ eventname: 'modalclose' })`

At a guess, consumers are more likely to listen for this event than emitting it, which might be a case for favoring past-tense event names over action/instruction names.

Reply:
Good suggestions - I went with `on` prefixes.